server/container_create.go

```diff
@@ -62,6 +64,33 @@ func addOciBindMounts(sb *sandbox, containerConfig *pb.ContainerConfig, specgen
 	return nil
 }
 
+func addDevices(sb *sandbox, containerConfig *pb.ContainerConfig, specgen *generate.Generator) error {
+	sp := specgen.Spec()
+	for _, device := range containerConfig.GetDevices() {
+		d, err := devices.DeviceFromPath(device.HostPath, device.Permissions)
```

Comment:
Personal twitch.  I HATE one char variable names unless it's some very local iterator (used within the space of 3 or 4 lines).   Please consider changing 'd' to 'devinfo' or something more easily spotted/searched for when maintaining later on down the road.